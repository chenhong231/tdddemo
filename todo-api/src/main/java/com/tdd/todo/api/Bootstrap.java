package com.tdd.todo.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;


@SpringBootApplication(scanBasePackages = "com.tdd.todo")
@EnableJpaRepositories({"com.tdd.todo.core"})
@EntityScan({"com.tdd.todo.core"})
public class Bootstrap {

    private static ConfigurableApplicationContext context;

    public static void main(final String[] args) {
        context = SpringApplication.run(Bootstrap.class, args);
    }

    public static void shutdown() {
        context.close();
    }
}
