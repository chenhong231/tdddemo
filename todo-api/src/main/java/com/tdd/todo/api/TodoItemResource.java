package com.tdd.todo.api;

import com.google.common.base.Strings;
import com.tdd.todo.core.TodoIndexParameter;
import com.tdd.todo.core.TodoItem;
import com.tdd.todo.core.TodoItemService;
import com.tdd.todo.core.TodoParameter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

import java.net.URI;
import java.util.List;
import java.util.Optional;
import java.util.stream.Collectors;

@RestController
@RequestMapping("/todo-items")
public class TodoItemResource {

    private TodoItemService service;

    @Autowired
    public TodoItemResource(final TodoItemService service) {
        this.service = service;
    }

    @PostMapping
    public ResponseEntity addTodoItem(@RequestBody final AddTodoItemRequest request) {
        if (Strings.isNullOrEmpty(request.getContent())) {
            return ResponseEntity.badRequest().build();
        }

        final TodoParameter parameter = TodoParameter.of(request.getContent());
        final TodoItem todoItem = this.service.addTodoItem(parameter);

        final URI uri = ServletUriComponentsBuilder
                .fromCurrentRequest()
                .path("/{id}")
                .buildAndExpand(todoItem.getIndex())
                .toUri();
        return ResponseEntity.created(uri).build();
    }

    @PutMapping("/{id}")
    public ResponseEntity markAsDone(@PathVariable("id") final int id,
                                     @RequestBody final MarkAsDoneRequest request) {
        if (id <= 0) {
            return ResponseEntity.badRequest().build();
        }

        if (!request.isDone()) {
            return ResponseEntity.badRequest().build();
        }

        final Optional<TodoItem> todoItem = this.service.markTodoItemDone(TodoIndexParameter.of(id));

        if (todoItem.isPresent()) {
            return ResponseEntity.ok().build();
        }

        return ResponseEntity.badRequest().build();
    }

    @GetMapping
    public List<TodoItemResponse> list(@RequestParam(value = "all", defaultValue = "false") final boolean all) {
        final List<TodoItem> items = this.service.list(all);
        return items.stream().map(TodoItemResponse::new).collect(Collectors.toList());
    }
}
