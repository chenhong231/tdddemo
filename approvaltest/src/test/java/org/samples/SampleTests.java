package org.samples;

import org.approvaltests.Approvals;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SampleTests {

    @Test
    public void testNormalJunit() {
        assertEquals(5,5);
    }

    @Test
    public void testWithApprovalTests() {
        Approvals.verify("Hello World");
    }

    @Test
    public void testJson() {
        Person hero = new Person("nayne", "cobb", true, 38);
        JsonApprovals.verifyAsJson(hero);
    }
}
