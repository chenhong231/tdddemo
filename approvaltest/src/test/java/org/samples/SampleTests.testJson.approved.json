{
  "firstName": "nayne",
  "lastName": "cobb",
  "isMale": true,
  "age": 38
}